File: backend/dashboard-service/Backend.Dashboards.Api/Dto/IssueAssigneeRemovedEvent.cs
using System.Text.Json.Serialization;

namespace Backend.Shared.DTOs;

public class IssueAssigneeRemovedEvent
{
    [JsonPropertyName("projectId")]
    public long ProjectId { get; set; }

    [JsonPropertyName("issueId")]
    public long IssueId { get; set; }

    [JsonPropertyName("removedUserId")]
    public long RemovedUserId { get; set; }

    [JsonPropertyName("removerId")]
    public long RemoverId { get; set; }

    [JsonPropertyName("removedAtUtc")]
    public DateTimeOffset RemovedAtUtc { get; set; } = DateTimeOffset.UtcNow;
}

File: backend/issue-service/src/main/java/com/example/issueservice/dto/rabbit/IssueAssigneeRemovedEvent.java
package com.example.issueservice.dto.rabbit;

import com.example.issueservice.dto.models.Issue;

import java.time.Instant;

public record IssueAssigneeRemovedEvent(
        long projectId,
        long issueId,
        long removedUserId,
        long removerId,
        Instant removedAtUtc
) {
    public static IssueAssigneeRemovedEvent from(Issue issue, long removedUserId, long removerId) {
        return new IssueAssigneeRemovedEvent(
                issue.getProjectId(),
                issue.getId(),
                removedUserId,
                removerId,
                Instant.now()
        );
    }
}